inspire_records/api/base.py

```diff
@@ -24,8 +24,87 @@
 
 from __future__ import absolute_import, division, print_function
 
+import uuid
+
+from invenio_pidstore.models import PersistentIdentifier, RecordIdentifier
+from inspire_dojson.utils import strip_empty_values
+from inspire_schemas.api import validate as schema_validate
+from invenio_pidstore.errors import PIDDoesNotExistError
 from invenio_records_files.api import Record
+from invenio_db import db
 
 
 class InspireRecord(Record):
     """Inspire Record."""
+
+    pid_type = None
+
+    @staticmethod
+    def strip_empty_values(data):
+        return strip_empty_values(data)
+
+    @staticmethod
+    def mint(record_uuid, data):
+        pass
+
+    def validate(self):
+        schema_validate(self)
+
+    @classmethod
+    def get_uuid_from_pid_value(cls, pid_value, pid_type=None):
+        if not pid_type:
+            pid_type = cls.pid_type
+        pid = PersistentIdentifier.get(pid_type, pid_value)
+        return pid.object_uuid
+
+    @classmethod
+    def get_record_by_pid_value(cls, pid_value, pid_type=None):
+        if not pid_type:
+            pid_type = cls.pid_type
+        record_uuid = cls.get_uuid_from_pid_value(pid_value)
+        record = cls.get_record(record_uuid)
+        return record
+
+    @classmethod
+    def create(cls, data, id_=None, **kwargs):
+        id_ = id_ or uuid.uuid4()
+        data = cls.strip_empty_values(data)
+        with db.session.begin_nested():
+            cls.mint(id_, data)
+            record = super(InspireRecord, cls).create(data, id_=id_, **kwargs)
+        return record
+
+    def update(self, data):
+        with db.session.begin_nested():
+            super(InspireRecord, self).update(data)
+            self.commit()
```

Comment:
It's only a Question. Do we really want to hide commit in update() method? It may help to keep data consistent between db and ES but it could also increase errors/unwanted commits in the code. If commit will be added here it also should be added everywhere so we could forget about manual commits. (consistency)

Reply:
leftover :/